### ISSUE TITLE
Farmers not being at all efficient

### ISSUE THREAD

**Issue by JohnyDL** [TangoTek/TekTopia-Community#54]
**Describe the bug**
A farmer or team of farmers will randomly harvest crops from a fully grown field rather than being strategic

**To Reproduce**
Grow a crop field to fully mature and watch farmers harvest it

**Expected behavior**
One of the few things Vanilla did right with villagers is harvesting crops: Harvest Crop at feet, till soil, replant look at adjacent blocks if fully grown move there and repeat. There probably shouldn't be many opportunities in your mod but it shouldn't behave worse than vanilla

**Screenshots**
![2019-05-26_18 34 40](https://user-images.githubusercontent.com/5202568/58385998-5025a100-7ff1-11e9-859c-cd943579c7de.png)
![2019-05-26_19 55 56](https://user-images.githubusercontent.com/5202568/58385999-50be3780-7ff1-11e9-97be-531f0c893ace.png)
![2019-05-26_18 34 18](https://user-images.githubusercontent.com/5202568/58386000-50be3780-7ff1-11e9-9220-0b84e78c3951.png)
![2019-05-26_18 34 30](https://user-images.githubusercontent.com/5202568/58386001-50be3780-7ff1-11e9-80bc-3161b1a925a2.png)

** Server/Location **
Offline and Survival1, Bugsy Towers again

**Comment by pdewilde** [comment]
I have multiple farms and my farmers spends most of her time walking between them, even though there are plenty of finished crops in each. Villager should harvest crops near to them first!

**Comment by pdewilde** [comment]
6 minutes following the farmer. (sped up 4x for your sanity)

https://youtu.be/1y1cGcE5Xek

**Comment by JohnyDL** [comment]
More of my offline farms still being badly mannaged
![2019-05-30_20 56 57](https://user-images.githubusercontent.com/5202568/58660612-bfff9880-831d-11e9-89f5-12bf6e74a3fc.png)
![2019-05-30_20 57 04](https://user-images.githubusercontent.com/5202568/58660614-c0982f00-831d-11e9-8ba8-4b9d0e96f60b.png)
![2019-05-30_20 57 10](https://user-images.githubusercontent.com/5202568/58660615-c0982f00-831d-11e9-8e6e-9689196829d3.png)
![2019-05-30_20 57 18](https://user-images.githubusercontent.com/5202568/58660617-c0982f00-831d-11e9-8345-fa0a4a595a78.png)
![2019-05-30_20 57 24](https://user-images.githubusercontent.com/5202568/58660618-c0982f00-831d-11e9-8f16-ec80b5695284.png)

**Comment by paspinall42** [comment]
Yeah small farms didn't produce enough, larger farms didn't get harvested (they would rather till a single patch of dirt than harvest the entire rest of the grown field) so that also led to villagers with no food, even trying to produce the farms all equidistant didn't help as they just wander between the 4 farms doing nothing.  I suspect its because harvesting is their final priority even behind gathering sugarcane.
